localstack/services/s3/s3_starter.py

```diff
@@ -278,14 +279,18 @@ def s3_response_handle_range_header(self, request, headers, response_content):
     # Patch utils_is_delete_keys
     # https://github.com/localstack/localstack/issues/2866
     # https://github.com/localstack/localstack/issues/2850
+    # https://github.com/localstack/localstack/issues/3931
     utils_is_delete_keys_orig = s3bucket_path_utils.is_delete_keys
 
     def utils_is_delete_keys(request, path, bucket_name):
         return path == '/' + bucket_name + '?delete=' or utils_is_delete_keys_orig(request, path, bucket_name)
 
     def s3_response_is_delete_keys(self, request, path, bucket_name):
         if self.subdomain_based_buckets(request):
-            return is_delete_keys(request, path, bucket_name)
+            # Temporary fix until moto supports x-id and DeleteObjects (#3931)
+            query = self._get_querystring(request.url)
+            is_delete_keys_v3 = (query and ('delete' in query) and get_safe(query, '$.x-id.0') == 'DeleteObjects')
```

Comment:
Maybe simply leave `get_safe(query, '$.x-id.0') == 'DeleteObjects'`, since it's very unlikely to receive non-delete request with x-is=DeleteObjects 🤔